src/native/corehost/host_runtime_contract.h

```diff
@@ -23,13 +23,37 @@
 #define HOST_PROPERTY_PLATFORM_RESOURCE_ROOTS "PLATFORM_RESOURCE_ROOTS"
 #define HOST_PROPERTY_TRUSTED_PLATFORM_ASSEMBLIES "TRUSTED_PLATFORM_ASSEMBLIES"
 
+struct trusted_platform_assemblies
+{
+	uint32_t assembly_count;
+	char** basenames; /* Foo.dll */
+	char** assembly_filepaths; /* /blah/blah/blah/Foo.dll */
+};
+
+struct probing_lookup_paths 
+{
+	uint32_t dir_count;
+	char** dirs;
+};
+
+struct probing_path_properties
```

Comment:
Naming nit: I would not use "probing". One of the goals of the .NET Core redesign of hosting was to avoid spurious probing for files on disk. The current implementation is a bit half-way-there, but realistically that's more of a bug than feature. TPA is already on the no-probing plan. And native and resource files are actually also no-probing inside hostpolicy, we just condense that information into the list of paths when calling into CoreCLR. Now I don't know if this is the time to change it (it can be a breaking change for some apps), but at least in terms of naming I would like to be clear that "We don't want to probe in the runtime" (note that hostpolicy does some amount of probing, but it does it so that the runtime doesn't have to).

Reply:
Thanks! I anticipated having to rename things :-).

Reply:
I would like to add here that we have an existing structure called `probe_paths_t`. We have to make sure that whatever we end up naming our probing paths stuff is clear enough to be differentiated from `probe_paths_t`. To not make the code more complicated to understand :)